bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/internal/p2/engine/phases/AuthorityChecker.java

```diff
@@ -52,6 +54,13 @@ public class AuthorityChecker {
 	private static final Pattern HIERARCHICAL_URI_PATTERN = Pattern
 			.compile("((?:[^/:]+):(?://[^/]+|///|/)?)([^?#]*)([#?].*)?"); //$NON-NLS-1$
 
+	private static final int REQUEST_TIMEOUT_MS = Integer
```

Comment:
Instead of system properties, the best is to fetch these from the `IProvisioningAgent`, this has several advantages e.g. you can configure this per agent and it automatically fall back to system properties.